### PR TITLE
[rllib] Support batch norm layers

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

- Pass in the right `is_training` tensor to `build_layers_v2`
- Add the right update ops to apply grads automatically
- Example script in `test_batch_norm.py`

I don't think this will work for e.g., A3C which applies gradient updates separately, but it should work fine in the other execution modes. 

## Related issue number

Closes: https://github.com/ray-project/ray/issues/2023

### CODE REVIEW

**Comment by ericl** [comment]
Ping @richardliaw

File: python/ray/rllib/evaluation/tf_policy_graph.py [hunk 1]
```diff
@@ -151,7 +164,7 @@ def build_compute_actions(self,
             builder.add_feed_dict({self._prev_action_input: prev_action_batch})
         if self._prev_reward_input is not None and prev_reward_batch:
             builder.add_feed_dict({self._prev_reward_input: prev_reward_batch})
-        builder.add_feed_dict({self._is_training: is_training})
+        builder.add_feed_dict({self._is_training: False})
```
Comment (richardliaw) [inline python/ray/rllib/evaluation/tf_policy_graph.py]: where in the code is `_is_training` True?

**Review by richardliaw — COMMENTED** [review]
A few questions

File: python/ray/rllib/evaluation/tf_policy_graph.py (hunk 1 above)
Comment (ericl) [reply python/ray/rllib/evaluation/tf_policy_graph.py]: https://github.com/ray-project/ray/blob/dedd3a802a972314afbe3799bfda3e8c9b098911/python/ray/rllib/evaluation/tf_policy_graph.py#L227

**Review by richardliaw — APPROVED** [review]

**Pull request merged**